### PR TITLE
Document Guild Member Timeout

### PR DESCRIPTION
**Pull request by lts20050703**
#1857 

update ModifyGuildMember, GuildMember, and GuildMemberUpdate interfaces

update BitwisePermissionFlags enum

### CODE REVIEW

File: src/types/guilds/modifyGuildMember.ts [hunk 1]
```diff
@@ -10,4 +10,6 @@ export interface ModifyGuildMember {
   deaf?: boolean | null;
   /** Id of channel to move user to (if they are connected to voice). Requires the `MOVE_MEMBERS` permission */
   channelId?: bigint | null;
+  /** when the user's timeout will expire and the user will be able to communicate in the guild again (up to 28 days in the future), set to null to remove timeout. Requires the `MODERATE_MEMBERS` permission */
+  communication_disabled_until?: number;
```
Comment (Skillz4Killz) [inline src/types/guilds/modifyGuildMember.ts:14]:
```suggestion
  communicationDisabledUntil?: number;
```

Comment (Skillz4Killz) [reply src/types/guilds/modifyGuildMember.ts:14]: These are options that the user will provide, we do not want snake case from users.

File: src/types/members/guildMemberUpdate.ts [hunk 2]
```diff
@@ -10,14 +10,17 @@ export interface GuildMemberUpdate {
...
   premiumSince?: string | null;
-  /** Whether the user has not yet passed the guild's Membership Screening requirements */
-  pending?: boolean;
   /** whether the user is deafened in voice channels */
   deaf?: boolean;
   /** whether the user is muted in voice channels */
   mute?: boolean;
+  /** Whether the user has not yet passed the guild's Membership Screening requirements */
+  pending?: boolean;
+  /** when the user's [timeout](https://support.discord.com/hc/en-us/articles/4413305239191-Time-Out-FAQ) will expire and the user will be able to communicate in the guild again, null or a time in the past if the user is not timed out */
```
Comment (Skillz4Killz) [inline src/types/members/guildMemberUpdate.ts:25]: you forgot to type the actual prop u only added the comment here

**Review by Skillz4Killz — CHANGES_REQUESTED** [review]

File: src/types/members/guildMemberUpdate.ts (hunk 2 above)
Comment (lts20050703) [reply src/types/members/guildMemberUpdate.ts:25]: oops my bad

File: src/types/guilds/modifyGuildMember.ts (hunk 1 above)
Comment (lts20050703) [reply src/types/guilds/modifyGuildMember.ts]: oops my bad too

**Comment by lts20050703** [comment]
@Skillz4Killz fixed. Feel free to look thoroughly

**Review by Skillz4Killz — APPROVED** [review]

**Pull request merged**